control-plane/pkg/reconciler/channel/channel_test.go

```diff
@@ -0,0 +1,270 @@
+      Retention: 604800000000000  # 1 week
+  Producer:
+    Idempotent: true  # Must be false for Azure EventHubs
+    RequiredAcks: -1  # -1 = WaitForAll, Most stringent option for "at-least-once" delivery.
+`,
+	"eventing-kafka": `
+cloudevents:
+   maxIdleConns: 1000
+   maxIdleConnsPerHost: 100
+kafka:
+  authSecretName: kafka-cluster
+  authSecretNamespace: knative-eventing
+  brokers: kafka:9092
+channel:
+  adminType: kafka # One of "kafka", "azure", "custom"
+  dispatcher:
+    cpuRequest: 100m
+    memoryRequest: 50Mi
+  receiver:
+    cpuRequest: 100m
+    memoryRequest: 50Mi
+`,
+}
+
+const (
+	//TODO: remove
+	ExpectedTopicDetail = "expectedTopicDetail"
+
+	finalizerName = "kafkachannels.messaging.knative.dev"
+)
+
+var finalizerUpdatedEvent = Eventf(
+	corev1.EventTypeNormal,
+	"FinalizerUpdate",
+	fmt.Sprintf(`Updated %q finalizers`, ChannelName),
+)
+
+var DefaultEnv = &config.Env{
+	DataPlaneConfigMapNamespace: "knative-eventing",
+	DataPlaneConfigMapName:      "kafka-channel-channels-subscriptions",
+	GeneralConfigMapName:        "kafka-broker-config",
+	IngressName:                 "kafka-channel-ingress",
+	SystemNamespace:             "knative-eventing",
+	DataPlaneConfigFormat:       base.Json,
+}
+
+// TODO: tests with and without subscriptions
+// TODO: tests for things from config-kafka (partition count, etc)
+// TODO: are we setting a InitialOffsetsCommitted status?
+
+func TestReconcileKind(t *testing.T) {
+
+	messagingv1beta.RegisterAlternateKafkaChannelConditionSet(base.IngressConditionSet)
+
+	env := *DefaultEnv
+	testKey := fmt.Sprintf("%s/%s", ChannelNamespace, ChannelName)
+
+	table := TableTest{
+		{
+			Name: "Reconciled normal - no subscription - no auth",
+			Objects: []runtime.Object{
+				NewChannel(),
+				NewService(),
+				ChannelReceiverPod(env.SystemNamespace, map[string]string{
+					base.VolumeGenerationAnnotationKey: "0",
+					"annotation_to_preserve":           "value_to_preserve",
+				}),
+				ChannelDispatcherPod(env.SystemNamespace, map[string]string{
+					base.VolumeGenerationAnnotationKey: "0",
+					"annotation_to_preserve":           "value_to_preserve",
+				}),
+				NewConfigMapWithTextData(system.Namespace(), constants.SettingsConfigMapName, configKafka),
+			},
+			Key: testKey,
+			WantUpdates: []clientgotesting.UpdateActionImpl{
+				ConfigMapUpdate(&env, &contract.Contract{
+					Generation: 1,
+					Resources: []*contract.Resource{
+						{
+							Uid:              ChannelUUID,
+							Topics:           []string{ChannelTopic()},
+							BootstrapServers: ChannelBootstrapServers,
+							Ingress: &contract.Ingress{
+								IngressType: &contract.Ingress_Path{
+									Path: receiver.Path(ChannelNamespace, ChannelName),
+								},
+							},
+							// TODO: do we need to set this?
+							//Auth:      &contract.Resource_AbsentAuth{},
+							// TODO: do we need to set this?
+							//Reference: &contract.Reference{Namespace: ChannelNamespace, Name: ChannelName},
```

Comment:
Do we need to set these for the channel? I see they're not set in broker+sink but set in source

Reply:
No, there is no need for them.